### ISSUE TITLE
Pushing the config file and certs to MongoDB fails when using the wildcard symbol

### ISSUE THREAD

**Issue by whalehub** [Ylianst/MeshCentral#107]
In the MeshCentral user guide, it says that you can import your config file and certs to your MongoDB using the following command:

```
node ./node_modules/meshcentral --dbpushconfigfiles * --configkey mypassword
```

I tried this out on my Linux machine just now and encountered the following error when using the wildcard symbol (*) in the command:

```
root@debian:~# node /opt/meshcentral/node_modules/meshcentral --dbpushconfigfiles * --configkey <redacted>
{ Error: Command failed: /usr/bin/node /opt/meshcentral/node_modules/meshcentral --dbpushconfigfiles docker-compose go openvpn shell-scripts --configkey <redacted> --launch
fs.js:119
    throw err;
    ^

Error: ENOENT: no such file or directory, open 'docker-compose/agentserver-cert-private.key'
    at Object.openSync (fs.js:448:3)
    at Object.readFileSync (fs.js:348:35)
    at obj.fs.readdir (/opt/meshcentral/node_modules/meshcentral/meshcentral.js:291:131)
    at FSReqCallback.args [as oncomplete] (fs.js:145:20)

    at ChildProcess.exithandler (child_process.js:297:12)
    at ChildProcess.emit (events.js:202:15)
    at maybeClose (internal/child_process.js:984:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:265:5)
  killed: false,
  code: 1,
  signal: null,
  cmd:
   '/usr/bin/node /opt/meshcentral/node_modules/meshcentral --dbpushconfigfiles docker-compose go openvpn shell-scripts --configkey <redacted> --launch' }
ERROR: MeshCentral failed with critical error, check MeshErrors.txt. Restarting in 5 seconds...
```

This is the output from mesherrors.txt:

```
-------- 2/22/2019, 5:47:11 AM ---- 0.2.9-a --------

fs.js:119
    throw err;
    ^

Error: ENOENT: no such file or directory, open 'docker-compose/agentserver-cert-private.key'
    at Object.openSync (fs.js:448:3)
    at Object.readFileSync (fs.js:348:35)
    at obj.fs.readdir (/opt/meshcentral/node_modules/meshcentral/meshcentral.js:291:131)
    at FSReqCallback.args [as oncomplete] (fs.js:145:20)
```

The same command works fine when I specify the absolute folder path.

```
root@debian:~# node /opt/meshcentral/node_modules/meshcentral --dbpushconfigfiles /opt/meshcentral/meshcentral-data --configkey <redacted>
Pushing agentserver-cert-private.key, 2498 bytes.
Pushing agentserver-cert-public.crt, 1534 bytes.
Pushing config.json, 611 bytes.
Pushing mpsserver-cert-private.key, 1702 bytes.
Pushing mpsserver-cert-public.crt, 1464 bytes.
Pushing root-cert-private.key, 2494 bytes.
Pushing root-cert-public.crt, 1546 bytes.
Pushing webserver-cert-private.key, 2494 bytes.
Pushing webserver-cert-public.crt, 1640 bytes.
Done.
```

**Comment by Ylianst** [comment]
Thanks for the detailed report. I will take a look at this next time I get a chance.

**Comment by Ylianst** [comment]
This issue should have been fixed about a week ago. If you can confirm & close that would be great. If you find any more problem, please it's hesitate to post it.

**Comment by Ylianst** [comment]
Any confirmation this issue is fixed? (It should be).

**Comment by whalehub** [comment]
@Ylianst I just gave it another try and it was unsuccessful again. Here's the output from the terminal:

```
root@debian:~# node /opt/meshcentral/node_modules/meshcentral --dbpushconfigfiles * --configkey <redacted>
No config.json in folder docker-compose
```

This time, there was no output in mesherrors.txt.

**Comment by Ylianst** [comment]
This is weird. When using *, it should replace that with the path to the "meshcentral-data" folder. However, in your case it replaced it with "docker-compose" which is not a valid path and so, could not find the config.json file at that location. Looking at the code, I don't know how this can happen. If you have any hints, that would be great.

FYI. In the next version v0.3.0-d, you don't need to use * anymore. If you don't specify a path after --dbpushconfigfiles, it will use the default "meshcentral-data" folder path.

**Comment by whalehub** [comment]
@Ylianst I can confirm that the issue has been fixed with v0.3.0-d. When I leave out the *, the files are pushed to the MongoDB database successfully.

**Comment by whalehub** [comment]
@Ylianst As for the reason why it tried to use the path "docker-compose", all I can really say about that is that "docker-compose" is the name of a folder in my user's home directory. I'm not sure why it would pick that one when * is specified in the command.
